src/domain/users/ip-metadata-validator.ts

```diff
@@ -0,0 +1,33 @@
+import { MissingIpMetadataError, InvalidIpMetadataTypeError } from "@domain/errors"
+import { getBlacklistedASNs, getWhitelistedCountries } from "@config"
+
+export const IpMetadataValidator = (): IpMetadataValidator => {
+  const validate = (userIP: UserIPs): true | ApplicationError => {
+    const blacklistedASNs = getBlacklistedASNs()
+    const whitelistedCountries = getWhitelistedCountries()
```

Comment:
I don't think we should be using loaded terms like `blacklist` / `whitelist`. There are many OSS projects that avoid them and since there is no cost associated with it I would follow this standard.
`blockList`, `denyList` and `allowList` / `acceptList` are more descriptive alternatives.

Reply:
Very good point, I've just pushed a change for this.